libglusterfs/src/dict.c

```diff
@@ -1469,28 +1476,14 @@ dict_reset(dict_t *dict)
         goto out;
     }
 
+    int i = 0;
     LOCK(&dict->lock);
-    data_pair_t *curr = dict->members_list;
-    data_pair_t *next = NULL;
-
-    while (curr != NULL) {
-        next = curr->next;
-        data_unref(curr->value);
-        GF_FREE(curr->key);
-        if (curr == &dict->free_pair) {
-            dict->free_pair.key = NULL;
-        } else {
-            mem_put(curr);
-        }
-        curr = next;
-    }
 
-    for (int i = 0; i < dict->hash_size; i++)
+    dict_clear_data(dict);
+    for (; i < dict->hash_size; i++)
```

Comment:
What performance are we gaining by not mentioning i=0 in this for loop? 
I think we should have written it here.

Reply:
On my 1st commit, I declared `i` inside the loop and got the `error: 'for' loop initial declarations are only allowed in C99 mode` compilation error in smoke-centos7 job - https://build.gluster.org/job/gh_smoke-centos7/766/console.
I mv both declaration+initialization outside the loop just for fixing the error.
will move the initialization inside the loop if this is the convention..